### ISSUE TITLE
Favicon is not showing if it's stored on remote URL and privacy plugin enabled

### ISSUE THREAD

**Issue by Guts** [squidfunk/mkdocs-material#4631]
### Contribution guidelines

- [X] I've read the [contribution guidelines](https://github.com/squidfunk/mkdocs-material/blob/master/CONTRIBUTING.md) and wholeheartedly agree

### I've found a bug and checked that ...

- [X] ... the problem doesn't occur with the `mkdocs` or `readthedocs` themes
- [X] ... the problem persists when all overrides are removed, i.e. `custom_dir`, `extra_javascript` and `extra_css`
- [X] ... the documentation does not mention anything about my problem
- [X] ... there are no open or closed issues that are related to my problem

### Description

When privacy plugin is enabled and the theme:favicon is pointing to a remote image, the favicon is not showing.

### Expected behaviour

rel=icon pointing to the favicon defined in theme configuration. Or specify that only local files can be used.

But I'm also asking myself about the case where the remote favicon is stored in one an excluded domain in privacy plugin. What to expect?

### Actual behaviour

Privacy plugin + remote favicon: no rel=icon

### Steps to reproduce

See configurations 

### Package versions

- Python: 3.10 (default on Ubuntu 22.04) but also on GitHub Actions with ubuntu-latest
- MkDocs: 1.4.2
- Material: 8.5.10-insiders-4.26.2

#### Without privacy plugin with local favicon file:

```yaml
site_name: My Docs

theme:
  name: material
  favicon: assets/images/geotribu_logo_64x64.png
```

Result: `<link rel="icon" href="assets/images/geotribu_logo_64x64.png">`

#### Without privacy plugin with remote file:

```yaml
site_name: My Docs

theme:
  name: material
  favicon: https://cdn.geotribu.fr/img/internal/charte/geotribu_logo_64x64.png

plugins:
  - privacy:
      enabled: false
```

Result : `<link rel="icon" href="https://cdn.geotribu.fr/img/internal/charte/geotribu_logo_64x64.png">`

#### With privacy plugin enabled and favicon as remote file:

```yaml
site_name: My Docs

theme:
  name: material
  favicon: https://cdn.geotribu.fr/img/internal/charte/geotribu_logo_64x64.png
  font: false

plugins:
  - privacy:
      enabled: true
      external_links_attr_map:
        target: _blank
      external_assets_exclude:
        # geotribu
        - cdn.geotribu.fr/images/*
```

Result: no favicon

### System information

- Operating system: every
- Browser: FF, Chromium...

**Comment by squidfunk** [comment]
Thanks for reporting. Fixed in squidfunk/mkdocs-material-insiders@5a08378a7. It didn't came to my mind that favicons might also be external. There's one more issue that doesn't impact functionality, but is a little odd – the file extension is sometimes appended albeit the file already includes the extension, i.e. `*.png.png`. I'll tackle this separately.

**Comment by squidfunk** [comment]
Released as part of [8.5.10+insiders-4.26.3](https://github.com/squidfunk/mkdocs-material-insiders/releases/tag/8.5.10-insiders-4.26.3).

I'll revisit the double extension thing in the coming weeks, as it doesn't impact functionality.

**Comment by squidfunk** [comment]
The problem with the redundant file extension was fixed in [`4290765`](squidfunk/mkdocs-material-insiders@429076514c5c4790d373885fe8246f1a190ba9c1) and released as part of [8.5.10+insiders-4.26.4](https://github.com/squidfunk/mkdocs-material-insiders/releases/tag/8.5.10-insiders-4.26.4).
